### ISSUE TITLE
"casting u32 to f64 may become silently lossy if types change" doesn't make sense

### ISSUE THREAD

**Issue by hrydgard** [rust-lang/rust-clippy#3964]
I'm pretty sure this warning should be removed. Same with f32 to f64, s32 to f64 and so on. All integer types with sizes <= u32 are trivially covered by f64, due to the 53 bit mantissa, and f32 should be too, right?

To me, having to do f64::from() is a lot visually noisier than "as f64" and I see no benefit in warning for something that will never occur in practice. I don't see any scenario in which the definition of f64 would change...

**Comment by oli-obk** [comment]
It's not that the definition of `f64` will change. It's also not the fact that converting an `u32` to a `f64` is ever lossy. It's about the fact that your source type might change (e.g. if you change the argument type from `u32` to `u64` in the following function)

```rust
fn foo(u: u32) {
    // loads of code here
    let x = u as f64;
    // use x
}
```

The conversion of `u32` to `f64` is not lossy here, but if the type of `u` is changed later, then you can get a lossy cast without realizing it, because everything still compiles.

**Comment by hrydgard** [comment]
I see. Then I guess my complains is that the wording of the warning is unclear. "If types will change" suggests the agency of the compiler, "if you later change the type" or something like that would be better I think.
